components/esp32_ble_presense/esp32_ble_presense.cpp

```diff
@@ -0,0 +1,139 @@
+////////////////////////////////////////////////////////////////////////
+// SPDX-License-Identifier: AGPL-3.0-only
+// (C) Copyright 2022-2024 - formatBCE
+
+#include "esp32_ble_presense.h"
+
+#include "esphome/core/log.h"
+
+#include <Arduino.h>
+#include <NimBLEDevice.h>
+
+#define bleScanInterval 0x80 // Used to determine antenna sharing between Bluetooth and WiFi. Do not modify unless you are confident you know what you're doing
+#define bleScanWindow 0x40 // Used to determine antenna sharing between Bluetooth and WiFi. Do not modify unless you are confident you know what you're doing
+
+using namespace esphome;
+
+namespace ESP32_BLE_Presense {
+
+NimBLEScan* pBLEScan;
+
+class BleAdvertisedDeviceCallbacks: public BLEAdvertisedDeviceCallbacks {
```

Comment:
I pulled this out of an inner class to keep the NimBLE headers out of the main header.